### ISSUE TITLE
zsh: forking process results in strange warning

### ISSUE THREAD

**Issue by Neo-Oli** [termux/termux-packages#1151]
```
$ ls &
[1] 21926
$ zsh: nice(5) failed: success
file1  file2  file3

[1]  + done       ls
```
I am not sure when this started happening, possibly when I upgraded to Android 7.1.1, or after that big recent update.

**Comment by Neo-Oli** [comment]
I found a workaround for this. Adding `unsetopt BG_NICE` to my .zshrc removes the warnings. 
Manual:

> BG_NICE (-6) <C> <Z>
> 
>     Run all background jobs at a lower priority. This option is set by default.

**Comment by xeffyr** [comment]
On Android 6.0.1 zsh forking works without warning.

**Comment by fornwall** [comment]
@Neo-Oli I cannot reproduce this problem on Android 7.1.2 on a Nexus 6p device.

Does `nice` work for you at all, like in `nice -n 2 ls`?

**Comment by Neo-Oli** [comment]
@fornwall. Yeah, nice works as expected (I even checked with htop). Zsh's nice function even works setting the niceness level too. It's just that it for some reason prints a warning even though nothing went wrong.

**Comment by fornwall** [comment]
Ok, I can actually reproduce it on Android 7.1.

The problem seems to be that Android now runs Termux processes with a niceness=-10, while Android 6 used a niceness=0.

Zsh adds five to the niceness level and checks if the return value of `nice(2)` is negative and if so believes that the call has failed. This is an incorrect check as noted by http://man7.org/linux/man-pages/man2/nice.2.html - errno should be used instead.

This has now been patched in version 5.3.1-3 of the zsh package. @Neo-Oli Can you confirm that it works now?

**Comment by Neo-Oli** [comment]
Seems to work now. cool \o/ Thanks
